kore/src/Kore/Internal/TermLike/TermLike.hs

```diff
@@ -798,6 +800,13 @@ traverseVariablesF adj =
 extractAttributes :: TermLike variable -> Attribute.Pattern variable
 extractAttributes (TermLike (attrs :< _)) = attrs
 
+markDefined :: TermLike variable -> TermLike variable
```

Comment:
This function should not exist. The `defined` attribute should only be constructed through its `Synthetic` instance. Otherwise, it isn't safe to modify the `TermLike` after marking it defined.